### PR TITLE
vimcal app

### LINKED ISSUE #5785: [CAL-502] vimcal app
null

<sub>[CAL-502](https://linear.app/calcom/issue/CAL-502/vimcal-app)</sub>

### PR DESCRIPTION
**Pull request by PeerRich**
CleanShot 2022-12-14 at 13 59 13

TODO:

- [ ] dont show calendar apps with external redirect in /getting-started onboarding calendar step

### CODE REVIEW

File: apps/web/pages/getting-started/[[...step]].tsx [hunk 1]
```diff
@@ -112,6 +112,7 @@ const OnboardingPage = (props: IOnboardingPageProps) => {
             <StepCard>
               {currentStep === "user-settings" && <UserSettings user={user} nextStep={() => goToIndex(1)} />}
 
+              {/* TODO: filter out calendar apps that have external links */}
               {currentStep === "connected-calendar" && <ConnectedCalendars nextStep={() => goToIndex(2)} />}
```
Comment (PeerRich) [inline apps/web/pages/getting-started/[[...step]].tsx:117]: todo: we wanna filter out calendar apps with external links from getting started
- vimcal
- cron
- amie.so

File: packages/app-store/vimcal/api/add.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  // Instead of passing appType and slug from here, api/integrations/[..args] should be able to derive and pass these directly to createCredential
+  appType: appConfig.type,
+  variant: appConfig.variant,
+  slug: appConfig.slug,
+  supportsMultipleInstalls: false,
+  handlerType: "add",
+  redirect: {
+    newTab: true,
+    url: "https://cal.com/blog/cal-plus-vimcal",
+  },
```
[the remark this answers is not in the document]
Comment (PeerRich) [reply packages/app-store/vimcal/api/add.ts:16]: see https://github.com/calcom/cal.com/pull/5791#discussion_r1048443517

File: apps/web/pages/getting-started/[[...step]].tsx (hunk 1 above)
Comment (PeerRich) [reply apps/web/pages/getting-started/[[...step]].tsx:117]: @hariombalhara will look into this

File: apps/web/components/apps/App.tsx [hunk 3]
```diff
@@ -66,7 +66,9 @@ const Component = ({
     }
   );
 
-  const allowedMultipleInstalls = categories.indexOf("calendar") > -1;
+  // variant not other allows, an app to be shown in calendar category without requiring an actual calendar connection e.g. vimcal
+  // Such apps, can only be installed once.
+  const allowedMultipleInstalls = categories.indexOf("calendar") > -1 && variant !== "other";
```
Comment (hariombalhara) [inline apps/web/components/apps/App.tsx:71]: cc @leog for review

File: apps/web/pages/getting-started/[[...step]].tsx [hunk 4]
```diff
@@ -112,6 +112,7 @@ const OnboardingPage = (props: IOnboardingPageProps) => {
             <StepCard>
               {currentStep === "user-settings" && <UserSettings user={user} nextStep={() => goToIndex(1)} />}
 
+              {/* TODO: filter out calendar apps that have external links */}
```
Comment (hariombalhara) [inline apps/web/pages/getting-started/[[...step]].tsx:115]:
```suggestion
```

**Review by hariombalhara — APPROVED** [review]
Looks good.

**Pull request merged**